source/custom/xnetsocket_converters.h

```diff
@@ -149,5 +149,68 @@ inline TimeValHolder convert_from_grpc(const pb_::Duration& input)
 {
   return TimeValHolder(input);
 }
+
+struct SockOptDataHolder {
+  SockOptData opt_data;
+  int32_t data_int;
+  bool data_bool;
+  std::string data_string;
+
+  SockOptDataHolder(const SockOptData& input) : opt_data(input)
+  {
+    data_int = opt_data.data_int32();
+    data_bool = opt_data.data_bool();
+    data_string = std::string(opt_data.data_string());
+  }
+
+  int32_t name() const
+  {
+    return opt_data.opt_name();
+  }
+
+  void* data()
+  {
+    switch (opt_data.data_case()) {
+      case SockOptData::DataCase::kDataInt32:
+        return &data_int;
+        break;
+      case SockOptData::DataCase::kDataBool:
+        return &data_bool;
+        break;
+      case SockOptData::DataCase::kDataString:
+        return &data_string[0];
+        break;
+      case SockOptData::DataCase::DATA_NOT_SET:
+        return nullptr;
+        break;
+    }
+  }
+
+  // size() method is used to simplify codegen calculating the size of the
+  // populated data.
+  nxsocklen_t size() const
+  {
+    switch (opt_data.data_case()) {
+      case SockOptData::DataCase::kDataInt32:
+        return sizeof(int32_t);
+        break;
+      case SockOptData::DataCase::kDataString:
+        return opt_data.data_string().size();
+        break;
+      case SockOptData::DataCase::kDataBool:
+        return sizeof(bool);
+      default:
+        return 0;
+        break;
+    }
+  }
+};
+
+template <typename TSockOptData>
+inline SockOptDataHolder convert_from_grpc(const SockOptData& input)
```

Comment:
nit: change this to SockOptDataInputConverter

Reply:
Done.